.ci/error-prone-check.groovy

```diff
@@ -0,0 +1,340 @@
+import groovy.transform.EqualsAndHashCode
+import groovy.transform.Field
+import groovy.transform.Immutable
+import groovy.xml.XmlUtil
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+@Field final static String SEPARATOR = System.getProperty("file.separator")
+@Field final static Set<String> profiles = Set.of("compile", "test-compile")
+
+final int exitCode
+if (args.length == 2) {
+    exitCode = parseArgumentAndExecute(args[0], args[1])
+}
+else {
+    exitCode = parseArgumentAndExecute(args[0], null)
+}
+System.exit(exitCode)
+
+/**
+ * Parse the command line arguments passed in and execute the branch based on the arguments.
+ *
+ * @param argument command line argument
+ * @return {@code 0} if command executes successfully, {@code 1} otherwise
+ */
+private int parseArgumentAndExecute(String argument, String flag) {
+    final String usageString = "Usage groovy ./.ci/error-prone-check.groovy" +
+            " (compile | test-compile) [-g | --generate-suppression]"
```

Comment:
Please wrap operator as we do for java code, `+ " ` then text in all scripts and java code. Ex: `+ "  (compile | test-compile)`. Also explain compile/ test-compile difference in javadoc.

Reply:
Can't be done in groovy, it does not use `;` to delimit statements, if we wrap `+` onto the next line, it won't compile.

Reply:
Explained compile and test-compile in javadoc